mcs/class/System.Private.CoreLib/System.Collections.Generic/EqualityComparer.cs

```diff
@@ -42,26 +42,16 @@ static EqualityComparer<T> CreateComparer ()
 			}
 
 			if (t.IsEnum) {
-				TypeCode underlyingTypeCode = Type.GetTypeCode(Enum.GetUnderlyingType(t));
-
 				return (EqualityComparer<T>)RuntimeType.CreateInstanceForAnotherGenericParameter (typeof(EnumEqualityComparer<>), t);
 			}
+			
 			return new ObjectEqualityComparer<T>();
 		}
 	}
 
 	partial class EnumEqualityComparer<T>
 	{
-        public override bool Equals(T x, T y) {
-            int x_final = System.Runtime.CompilerServices.JitHelpers.UnsafeEnumCast(x);
-            int y_final = System.Runtime.CompilerServices.JitHelpers.UnsafeEnumCast(y);
-            return x_final == y_final;
-        }
-		/*
-        public override int GetHashCode(T obj) {
-            int x_final = System.Runtime.CompilerServices.JitHelpers.UnsafeEnumCast(obj);
-            return x_final.GetHashCode();
-        }
```

Comment:
This is going to be needed, the version in shared/ does boxing.

Reply:
Why? It calls it using constrained. !T